### PR TITLE
support rtp/rtcp over tcp

### PR DESCRIPTION
**Pull request by mmaatuq**
Please sign (check) the below before submitting the Pull Request:

- [x] I have signed the ntop Contributor License Agreement at https://github.com/ntop/legal/blob/main/individual-contributor-licence-agreement.md
- [x] I have read the contributing guide lines at https://github.com/ntop/nDPI/blob/dev/CONTRIBUTING.md
- [x] I have updated the documentation (in doc/) to reflect the changes made (if applicable)

Link to the related [issue](https://github.com/ntop/nDPI/issues/2422):

Describe changes:
*  support rtp/rtcp over tcp as per rfc4571. 
*  will be in draft for now so i can add test case and fix other tests.

### CODE REVIEW

**Comment by mmaatuq** [comment]
regarding 
```C
  /* NDPI_PROTOCOL_RTP */
   u_int32_t rtp_stage:2;
   u_int8_t rtp_seq_set[2];
   u_int16_t rtp_seq[2];

   /* NDPI_PROTOCOL_RTCP */ 
   u_int32_t rtcp_stage:2;
```
I'm not so sure if declaring them direct inside ndpi_flow struct  is the right place to define them 
I tried to make them part of the protos union, 
but this caused some issues discovered during fuzz testing,
 for example when we have rtp_stage  = 1
but after this the protocol identified as tls,quic 
and it doesn't have server_names but as rtp_stage is set 
the below check  passed 
```C
      if(flow->protos.tls_quic.server_names)
	ndpi_free(flow->protos.tls_quic.server_names);
```
which caused segfault.

also, i'm not completely sure if that was the reason for trying to free an object that wasn't allocated, the above explanation is my analysis.

File: src/include/ndpi_typedefs.h [hunk 1]
```diff
@@ -1440,6 +1430,9 @@ struct ndpi_flow_struct {
       u_int8_t url_count;
       char url[4][48];
     } slp;
+    
+
+    
   } protos;
```
Comment (IvanNardi) [inline src/include/ndpi_typedefs.h]: Remove these 3 new lines, please

File: src/include/ndpi_typedefs.h [hunk 2]
```diff
@@ -1498,6 +1491,14 @@ struct ndpi_flow_struct {
...
   /* NDPI_PROTOCOL_TINC */
   u_int8_t tinc_state;
+ 
+   /* NDPI_PROTOCOL_RTP */
+   u_int32_t rtp_stage:2;
+   u_int8_t rtp_seq_set[2];
+   u_int16_t rtp_seq[2];
+   
+   /* NDPI_PROTOCOL_RTCP */ 
+   u_int32_t rtcp_stage:2;
```
Comment (IvanNardi) [inline src/include/ndpi_typedefs.h]: u_int8_t and move it above RTP

File: src/include/ndpi_typedefs.h [hunk 3]
```diff
@@ -1498,6 +1491,14 @@ struct ndpi_flow_struct {
 
   /* NDPI_PROTOCOL_TINC */
   u_int8_t tinc_state;
+ 
+   /* NDPI_PROTOCOL_RTP */
+   u_int32_t rtp_stage:2;
```
Comment (IvanNardi) [inline src/include/ndpi_typedefs.h]: u_int8_t

File: src/lib/protocols/rtp.c [hunk 4]
```diff
@@ -259,7 +305,7 @@ void init_rtp_dissector(struct ndpi_detection_module_struct *ndpi_struct,
   ndpi_set_bitmask_protocol_detection("RTP", ndpi_struct, *id,
 				      NDPI_PROTOCOL_RTP,
 				      ndpi_search_rtp,
-				      NDPI_SELECTION_BITMASK_PROTOCOL_V4_V6_UDP_WITH_PAYLOAD,
+                      NDPI_SELECTION_BITMASK_PROTOCOL_V4_V6_TCP_OR_UDP_WITH_PAYLOAD_WITHOUT_RETRANSMISSION,
```
Comment (IvanNardi) [inline src/lib/protocols/rtp.c]: could you try to restore old indentation, please?

**Review by IvanNardi — CHANGES_REQUESTED** [review]
Overall it is quite good!
Just some minor issues below.
Furthermore:
* try to merge together the 2 RTP traces, if possible
* rebase

**Comment by IvanNardi** [comment]
> ```c
> flow->protos
> ```

You got it right: you can write to `flow->protos` union only after set a proper classification

**Comment by mmaatuq** [comment]
> Overall it is quite good! Just some minor issues below. Furthermore:
> 
>     * try to merge together the 2 RTP traces, if possible
> 
>     * rebase

for merging 2 rtp traces, [here](https://github.com/ntop/nDPI/files/15406435/rtp_tcp.zip) there are 3 traces 2 of them  have the same rtp flow and the third one has fragmented packets (that are not covered in this PR)

**Comment by IvanNardi** [comment]
> > Overall it is quite good! Just some minor issues below. Furthermore:
> > ```
> > * try to merge together the 2 RTP traces, if possible
> > 
> > * rebase
> > ```
> 
> for merging 2 rtp traces, [here](https://github.com/ntop/nDPI/files/15406435/rtp_tcp.zip) there are 3 traces 2 of them have the same rtp flow and the third one has fragmented packets (that are not covered in this PR)

@mmaatuq , sorry, I have not been clearer: I was asking to merge "rtp_tcp.pcapng" that you added to the PR and the existing "rtp.pcapng" already in the repository. This way we have only one RTP trace in the tests directory...

**Review by IvanNardi — APPROVED** [review]
@mmaatuq, thanks for this work

**Pull request merged**